config/configmapprovider/file.go

```diff
@@ -18,12 +18,17 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
+	"time"
 
 	"go.opentelemetry.io/collector/config"
 )
 
+var errNoOp = errors.New("no change")
+
 type fileMapProvider struct {
 	fileName string
+	watching bool
```

Comment:
What's the purpose of this? It does't seem to be set anywhere.

Reply:
fixed this one. this is to ensure that we create a watch only once as `Retrieve` is called each time the `onChange` is invoked during a file change.